flink-runtime/src/main/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolder.java

```diff
@@ -143,12 +144,14 @@ private OperatorCoordinatorHolder(
 
     public void lazyInitialize(
             GlobalFailureHandler globalFailureHandler,
-            ComponentMainThreadExecutor mainThreadExecutor) {
+            ComponentMainThreadExecutor mainThreadExecutor,
+            @Nullable CheckpointCoordinator checkpointCoordinator) {
 
         this.globalFailureHandler = globalFailureHandler;
         this.mainThreadExecutor = mainThreadExecutor;
+        context.lazyInitialize(globalFailureHandler, mainThreadExecutor, checkpointCoordinator);
 
-        context.lazyInitialize(globalFailureHandler, mainThreadExecutor);
+        context.lazyInitialize(globalFailureHandler, mainThreadExecutor, checkpointCoordinator);
```

Comment:
@lindong28 @yunfengzhou-hub Is initializing the `OperatorCoordinatorHolder` twice done deliberately here? :thinking:

Reply:
Thanks for clarification. Doing it as a hotfix commit in one other PR makes sense because of the reasons you mentioned. :+1: